pkg/api/types_test.go

```diff
@@ -416,3 +417,82 @@ loop:
 		t.Fatalf("mismatch between internal and admin API fields: afield=%q, ifield=%q", afield, ifield)
 	}
 }
+
+func Test20180930previewAPIParity(t *testing.T) {
+	// the algorithm is: list the fields of both types.  If the head of either
+	// list is expected not to be matched in the other, pop it.  If the heads of
+	// both lists match, pop them.  In any other case, error out.
+
+	pfields := walk(reflect.TypeOf(v20180930preview.OpenShiftManagedCluster{}),
+		map[string][]reflect.Type{
+			".Properties.AuthProfile.IdentityProviders.Provider": {
+				reflect.TypeOf(v20180930preview.AADIdentityProvider{}),
+			},
+		},
+	)
+	ifields := walk(reflect.TypeOf(OpenShiftManagedCluster{}),
+		map[string][]reflect.Type{
+			".Properties.AuthProfile.IdentityProviders.Provider": {
+				reflect.TypeOf(AADIdentityProvider{}),
+			},
+		},
+	)
+
+	// TODO: I don't believe this should be in the admin type at all
+	notInInternal := []*regexp.Regexp{
+		regexp.MustCompile(`^\.Response`),
+		regexp.MustCompile(`^\.Properties\.MasterPoolProfile`),
+		regexp.MustCompile(`^\.Properties\.AuthProfile\.IdentityProviders\.Provider\.Secret`),
+	}
+
+	// TODO: why don't we just include all of these in the admin type?
+	notInV20180930preview := []*regexp.Regexp{
+		regexp.MustCompile(`^\.Properties\.ServicePrincipalProfile`),
+		regexp.MustCompile(`^\.Properties\.AzProfile`),
```

Comment:
ditto

Reply:
Sorry, servicePrincipalProfile and azProfile are not included in the customer profile, disregard